---
title: "PILMBC211"
---

# Requirements

## knowledge 
- PILMBC 103: https://pilm-bioinformatics.github.io/pilmbc103-remote-storage-and-computing/
- What is RNAseq: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2949280/

## Software
- Docker: https://docs.docker.com/v17.12/install/
- [nexflow](https://www.nextflow.io/): In the terminal do this: 
    
    `java -version` This has to return a version > 8 or >1.8
    
    `curl -s https://get.nextflow.io | bash`
    
    `./nextflow run hello`

# Agenda

- 1:30-1:55 Introduction to _vim_ __hands-on__: modify files in the terminal (mandatory if you don't know this)
- 2:00-2:25 Best practices for RNAseq analysis __lecture__
- 2:30-3:50 Running an example in the local computer __hands-on__
- 3:00-3:30 Running an example in OpenMind cluster __hand-on__: you need to be part of the cluster, contact your representative:
   - millerlab : Jefferson Roy
   - surlab: Ming Hu
   - tsailab: Lorena Pantano
   
You are welcome to stay during the whole session even if you don't want/can't to follow the _hands-on_ materials.

# Goal

- Best practices on choosing the right tool
- How to process FASTQ files from RNAseq experiments
- Run a test data analysis in your computer or/and the Open Mind cluster

# Angels

If you see you can help others during the workshop, please do, it will improve the dynamic.

# Contributors

Happy to accept corrections at: https://github.com/pilm-bioinformatics/pilmbc211-rnaseq-i